### PR TITLE
Add IQ Testnet

### PR DESCRIPTION
**Pull request by SidharthK2**
# Integrate IQ test chain

- Replaces Goerli with IQ test chain

## How should this be tested?

1. Staking
2. Wallet drawer
3. Stats

## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/2649
closes https://github.com/EveripediaNetwork/issues/issues/2608

### CODE REVIEW

File: src/config/wagmi.ts [hunk 1]
```diff
@@ -2,20 +2,50 @@ import { WalletConnectConnector } from '@wagmi/core/connectors/walletConnect'
...
+  name: 'IQ Chain',
+  network: 'IQ Chain',
+  nativeCurrency: {
+    name: 'IQ Token',
+    symbol: 'IQ',
+    decimals: 18,
+  },
+  rpcUrls: {
+    default: { http: ['https://rpc-testnet.braindao.org/'] },
+    public: { http: ['https://rpc-testnet.braindao.org/'] },
```
Comment (kesar) [inline src/config/wagmi.ts:25]: i think we can just have default. not sure whats gonna make public different

File: src/hooks/useLock.ts [hunk 2]
```diff
@@ -9,10 +9,17 @@ const hiiqContractConfig = {
...
 
-const erc20ContractConfig = {
-  address: config.iqAddress as `0x${string}`,
-  abi: erc20Abi,
-}
+//use WIQ for dev
+const erc20ContractConfig =
+  config.alchemyChain === 'iqChain'
+    ? {
+        address: config.wiqAddress as `0x${string}`,
```
Comment (kesar) [inline src/hooks/useLock.ts:16]: if you are gonna be doing this if/else condition across all site, just do a config.erc20IQ in the config and use that. Then you will only have that condition there and code will be more clean

File: src/data/LockConstants.tsx [hunk 3]
```diff
@@ -1,12 +1,18 @@
 import config from '@/config'
 
+const DEV_GAS_LIMIT = 10_000_000
```
Comment (kesar) [inline src/data/LockConstants.tsx:3]: TESTNET better than DEV so we keep same naming for all testnet related

File: src/config/index.ts [hunk 4]
```diff
@@ -1,29 +1,30 @@
...
   infuraId: process.env.NEXT_PUBLIC_INFURA_ID,
   alchemyApiKey: String(process.env.NEXT_PUBLIC_ALCHEMY_API_KEY),
-  alchemyChain: process.env.NEXT_PUBLIC_ALCHEMY_CHAIN || 'goerli',
+  alchemyChain: process.env.NEXT_PUBLIC_ALCHEMY_CHAIN || 'iqChain',
   graphqlUrl:
     process.env.NEXT_PUBLIC_EP_API || 'https://api.dev.braindao.org/graphql',
-  chainId: process.env.NEXT_PUBLIC_CHAIN_ID || '5',
-  chainName: process.env.NEXT_PUBLIC_CHAIN_NAME || 'goerli',
+  chainId: process.env.NEXT_PUBLIC_CHAIN_ID || '313377',
+  chainName: process.env.NEXT_PUBLIC_CHAIN_NAME || 'iqchain',
```
Comment (kesar) [inline src/config/index.ts:27]: default to iqchain but comparison is w iqChain

File: src/config/wagmi.ts [hunk 5]
```diff
@@ -2,20 +2,50 @@ import { WalletConnectConnector } from '@wagmi/core/connectors/walletConnect'
...
+  },
+  blockExplorers: {
+    default: { name: 'BrainScan', url: 'https://testnet.braindao.org/' },
```
Comment (kesar) [inline src/config/wagmi.ts:28]: you can use config.blockExplorerUrl

Comment (SidharthK2) [reply src/config/wagmi.ts:28]: Noted ser

**Review by Tolu-Mals — APPROVED** [review]

**Comment by kesar** [comment]
code looks good. ensure you test that production works as expected after merging 👍🏻

**Comment by SidharthK2** [comment]
> code looks good. ensure you test that production works as expected after merging 👍🏻

Absolutely

**Pull request merged**